JsonLow.js

```diff
@@ -406,7 +406,7 @@ export const JsonLow = (next, initialState = {}) => {
     },
     state: () => {
       const downstream = next.state?.()
-      return {mode, parents: [...parents], isKey, downstream}
+      return {mode, parents: [...parents], isKey, downstream, hexIndex, maxDepth}
```

Comment:
I'm not sure about `maxDepth` here. Form a pragmatic point of view, it's fine.

From consistency/elegance/minimalism point of view it doesn't quite fit with the other props. It's the only one that remains constant for the entire lifetime of a `JsonLow` instance.

The construction site will always know `maxDepth` if it provides it. Otherwise it will be the default (65536 at the moment) -- which is the most problematic case, as you can't really obtain that default by any means. I suppose it could be declared as some constant that you can import.

Anyway, just a remark. I can accept the current pragmatic solution if it makes life easier, etc. So up to you.

Reply:
I agree with you, `maxDepth` feels awkward here. I added a new `config` method, which for now just returns `maxDepth`, and has a `downstream` property like the `state` method does. Let me know if you think this is a bad idea. I don't think exporting the default value is a good idea because a library that wraps the parser might want to change the defaults, which is why I ended up with this approach.

`downstream` is also at the end now